### ISSUE TITLE
Search  listeners on 0.2.0 

### ISSUE THREAD

**Issue by azazadev** [excogitatr/paginate_firestore#35]
Hello,

from changelog of 0.2.0 version

    Added support for Search and Refresh listeners

but when I'm looking on the implementation we throw UnimplementedError

```
  @override
  void initState() {
    _scrollController = widget.scrollController ?? ScrollController();
    if (widget.listeners != null) {
      for (var listener in widget.listeners) {
        if (listener is PaginateRefreshedChangeListener) {
          listener.addListener(() {
            if (listener.refreshed) {
              refresh();
            }
          });
        } else if (listener is PaginateSearchChangeListener) {
          listener.addListener(() {
            throw UnimplementedError();
          });
        }
      }
    }
```
so should we implement SearchChangeListener ? if yes can you please provide sample

Thanks

**Comment by excogitatr** [comment]
Hey @azazadev , I apologize for that mistake. It is now fixed in v0.2.1 and you can find the usage in this pr #34

**Comment by azazadev** [comment]
Many Thanks !!! will try to test it today
